### PR TITLE
feat: add `noir-compiler` checks to `aztec_macros`

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Problem\*

Resolves 

## Summary\*

This PR moves the checks applied in [noir_compiler](https://github.com/AztecProtocol/aztec-packages/blob/c38dbd2fe13dfcf019da3877c17ca8d148f45233/yarn-project/noir-compiler/src/contract-interface-gen/abi.ts#L92-L97) to `aztec_macros`.

## Documentation\*

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[Exceptional Case]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by Maddiaa0 — APPROVED** [review]
lgtm

File: aztec_macros/src/lib.rs [hunk 1]
```diff
@@ -31,10 +31,15 @@ impl MacroProcessor for AztecMacro {
     }
 }
 
+const FUNCTION_TREE_HEIGHT: u32 = 5;
+const MAX_CONTRACT_FUNCTIONS: usize = 2_usize.pow(FUNCTION_TREE_HEIGHT);
```
Comment (spalladino) [inline aztec_macros/src/lib.rs:35]: Should we have `circuits.js/src/scripts/constants.in.ts` spit out a .rs file in noir with this data?

Comment (Maddiaa0) [reply aztec_macros/src/lib.rs:35]: yes - very good point

Comment (Maddiaa0) [reply aztec_macros/src/lib.rs:35]: https://github.com/AztecProtocol/aztec-packages/issues/4129

**Pull request merged**